ArduCopter/system.cpp

```diff
@@ -380,8 +380,16 @@ void Copter::allocate_motors(void)
         case AP_Motors::MOTOR_FRAME_DECA:
         case AP_Motors::MOTOR_FRAME_SCRIPTING_MATRIX:
         default:
-            motors = new AP_MotorsMatrix(copter.scheduler.get_loop_rate_hz());
-            motors_var_info = AP_MotorsMatrix::var_info;
+#if defined(STM32H7) || CONFIG_HAL_BOARD != HAL_BOARD_CHIBIOS
+            if ((copter.g2.flight_options & uint32_t(FlightOptions::USE_OPTIMAL_MIXER)) != 0) {
+                motors = new AP_MotorsMatrix_Optimal(copter.scheduler.get_loop_rate_hz());
```

Comment:
might be worth using fast memory, with malloc_type()

Reply:
done, marginally faster